vcl/rules/generic.vcl

```diff
@@ -4,7 +4,7 @@
 sub vcl_recv {
     # Bad User-Agent - Scanners
     # - http://mod-security.svn.sourceforge.net/ (modsecurity_35_scanners.data)
-    if (req.http.User-Agent ~ "(i)(metis|bilbo|n-stealth|black widow|brutus|cgichk|webtrends security|jaascois|pmafind|\.nasl|nsauditor|paros|nessus|nikto|webinspect|blackwidow)") {
+    if (req.http.User-Agent ~ "(?i:(?:^(grabber|bsqlbf|mozilla\/4\.0 (compatible)|sqlmap|mozilla\/4\.0 (compatible; msie 6.0; win32)|mozilla\/5\.0 sf\/\/|arachni|sql power injector|absinthe|netsparker|python-httplib2|dirbuster|pangolin|nmap nse|sqlninja|grendel-scan|havij|w3af|hydra|nstalker|n-stalker|openvas|fimap|yanga|url_spider_sql|topblogsinfo|purebot|jikespider|google_three_web|aboundexbot|mozilla\/the mole|themole\.nasel\.com\.ar|metis|bilbo|n-stealth|black widow|brutus|cgichk|webtrends security|jaascois|pmafind|\.nasl|nsauditor|paros|nessus|nikto|webinspect|blackwidow)$))") {
```

Comment:
I'm curious if you've checked any of these - in particular, if you've seen the "mozilla/4.0 (compatible)" match? Parens are a special character in regex..

Reply:
@comotion You're right.

I actually copied this regex directly from the MedSecurity rules. This same rule is used globally by several hosting companies.

Despite incorrect parens, this should not affect the rule's functioning.
If you prefer, I can fix that.

Reply:
yes, fix it please.